File: windows/.agent/skills/review-case/references/incorrect-lifecycle.cpp
// ❌ INCORRECT: Creating engine on every join
void CExampleDlg::JoinChannel() {
    m_rtcEngine = createAgoraRtcEngine();  // WRONG: recreates engine
    m_rtcEngine->joinChannel(...);
}

// ❌ INCORRECT: Not calling release
void CExampleDlg::PostNcDestroy() {
    m_rtcEngine->leaveChannel();
    // Missing: m_rtcEngine->release()
    CDialogEx::PostNcDestroy();
}

File: macOS/.agents/skills/review-case/references/incorrect-lifecycle.swift
// ❌ INCORRECT: Creating engine on every join
func joinChannel() {
    agoraKit = AgoraRtcEngineKit(...)  // WRONG: recreates engine
    agoraKit.joinChannel(...)
}

// ❌ INCORRECT: Not calling destroy
override func viewWillBeRemovedFromSplitView() {
    agoraKit.leaveChannel(nil)
    // Missing: AgoraRtcEngineKit.destroy()
    super.viewWillBeRemovedFromSplitView()
}
